libs/dal/src/repositories/message/message.repository.ts

```diff
@@ -126,18 +128,41 @@ export class MessageRepository extends BaseRepository<MessageDBModel, MessageEnt
       requestQuery.archived = { $in: [true, false] };
     }
 
+    const snoozedCondition: Array<MessageQuery> = [];
     if (query.snoozed != null) {
       if (query.snoozed) {
         requestQuery.snoozedUntil = { $ne: null };
       } else {
-        requestQuery.$or = [{ snoozedUntil: { $exists: false } }, { snoozedUntil: null }];
+        snoozedCondition.push({ snoozedUntil: { $exists: false } }, { snoozedUntil: null });
+      }
+    }
+
+    const severityCondition: Array<MessageQuery> = [];
+    if (query.severity && query.severity?.length > 0) {
+      if (query.severity.includes(SeverityLevelEnum.NONE)) {
+        severityCondition.push({ severity: { $exists: false } }, { severity: { $in: query.severity } });
+      } else {
+        requestQuery.severity = { $in: query.severity };
```

Comment:
this is an improvement over many different or statements; so for example when I search for the `snoozed: false` and `severity: high` I want to get only that match both parts of the condition